services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -271,6 +276,133 @@ export class TransactionJobProcessorsService {
     // 5. No transaction stored or updated after API-call, because waiting transaction is already stored earlier and will remain 'waiting' at this stage (to be updated via callback)
   }
 
+  public async processNedbankTransactionJob(
+    transactionJob: NedbankTransactionJobDto,
+  ): Promise<void> {
+    // 1. Get the registration to log changes to it later in event table
+    const registration = await this.getRegistrationOrThrow(
+      transactionJob.referenceId,
+    );
+    const oldRegistration = structuredClone(registration);
+
+    // 2. Set the payment reference
+    // This is a unique identifier for each transaction, which will be shown on the bank statement which the user receives by Nedbank out of the 121-platform
+    // It's therefore a human readable identifier, which is unique for each transaction and can be related to the registration and transaction manually
+    // Payment reference cannot be longer than 30 characters
+    const paymentReferencePrefix =
+      (await this.programFinancialServiceProviderConfigurationRepository.getPropertyValueByName(
+        {
+          programFinancialServiceProviderConfigurationId:
+            transactionJob.programFinancialServiceProviderConfigurationId,
+          name: FinancialServiceProviderConfigurationProperties.paymentReferencePrefix,
+        },
+      )) as string; // This must be a string. If it is undefined the validation in payment service should have caught it. If a user set it as an array string you should get an internal server error here, this seems like an edge case;
+    const sanitizedPaymentReferencePrefix = paymentReferencePrefix.replace(
+      /[^a-zA-Z0-9]/g,
+      '',
+    ); // All non-alphanumeric characters are removed because the nedbank API does not accept them
```

Comment:
Plus dashes?
And should we document adding FspConfig and this behavior here?
https://github.com/global-121/121-platform/wiki/Nedbank